lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1445,9 +1449,25 @@ protected function _validateAndCompleteFieldMapping(array &$mapping)
      *
      * @param array $mapping The mapping.
      *
-     * @return array The updated mapping.
+     * @return mixed[] The updated mapping.
```

Comment:
Oh... now I remember why we started 3.0...
Maybe we should do something on psalm to ignore these return mappings and only enforce on 3.0. The entire 2.X series we are only relying on complex array structures as they were 1- more memory effective and 2- faster than objects.
This is no longer true on PHP 7.

Reply:
Glad to learn all these `mixed[]` will go away in 3.x :)